crates/sqlexec/src/context/local.rs

```diff
@@ -81,7 +82,8 @@ impl LocalSessionContext {
             .with_extension(Arc::new(catalog_mutator))
             .with_extension(Arc::new(native_tables.clone()))
             .with_extension(Arc::new(TempCatalog::default()));
-        let state = SessionState::with_config_rt(conf, Arc::new(runtime));
+        let state = SessionState::with_config_rt(conf, Arc::new(runtime))
+            .add_physical_optimizer_rule(Arc::new(RuntimeGroupPullUp {}));
```

Comment:
I didn't do this for `RemoteSessionContext` since from what I can see it only handles the actual execution of the physical plans, as opposed to creating/optimizing them like in the case of a `LocalSessionContext`.

On a related note, when testing this on the qa instance I've noticed that `SessionState` seems to be serialized from the remote and sent back to the local machine? This is because I don't see `RuntimeGroupPullUp` in the available optimizers inside of `Session::create_physical_plan`, even though it is there when the `Session` is created. If this returned `SessionState` originates from this part of the logic then the full end-to-end test can be done once this change is deployed to the remote GlareDB instance in the cloud. Otherwise I'd need to make an equivalent change to `RemoteSessionContext` as well.

Reply:
> I didn't do this for RemoteSessionContext since from what I can see it only handles the actual execution of the physical plans, as opposed to creating/optimizing them like in the case of a LocalSessionContext.

Yep, we're trying to keep the planning/optimization on the local side.

> On a related note, when testing this on the qa instance I've noticed that SessionState seems to be serialized from the remote and sent back to the local machine?

So you're probably hitting this issue because we're creating a new session state when we "attach" a remote session:

https://github.com/glaredb/glaredb/blob/244817e6394f31eccb1de98b1830a39b2c91c14c/crates/sqlexec/src/context/local.rs#L115-L131

It's not the cleanest right now, but was the easiest thing to do. I'm open to alternatives to creating the session states/contexts that make these things less likely to trip someone up.

Reply:
Oh makes sense. Yeah, I think that's fine for the time being, I just need to add our optimizer rule there too.